lib/readline.js

```diff
@@ -247,6 +248,10 @@ Interface.prototype._addHistory = function() {
   // if the history is disabled then return the line
   if (this.historySize === 0) return this.line;
 
+  const line = this.line.replace(/^\s+|\s+$/, '');
+  // if the trimmed line is empty then return the line
+  if (line.length === 0) return this.line;
+
```

Comment:
Yes. Incorporated test to ensure that whitespace-only lines are not added in history.